index.bs

```diff
@@ -287,12 +287,33 @@ Perform Logger("warn", <var>data</var>).
 
 <h3 id="grouping">Grouping methods</h3>
 
+A <dfn>group</dfn> is an implementation-specific, potentially-interactive view for output produced by calls to Printer, with one further level of indentation than its parent. Each {{console}} namespace object has an associated <dfn>group stack</dfn>, which is a <a>stack</a>, initially empty. Only
+the last <a>group</a> in a <a>group stack</a> will host output produced by calls to Printer.
+
 <h4 id="group" oldids="group-data,dom-console-group" method for="console">group(...<var>data</var>)</h4>
 
+<emu-alg>
+  1. Let _group_ be a new <a>group</a>.
+  1. Let _groupLabel_ be the result of Formatter(data), and incorporate _groupLabel_ as a label for _group_.
+  1. Optionally, if the environment supports interactive groups, _group_ should be expanded by default.
+  1. Perform Printer("log", «_group_»).
+  1. Push _group_ onto the appropriate <a>group stack</a>
```

Comment:
Missing period. Also you can do `<a>Push</a>` (maybe `<a for="stack">Push</a>` will be necessary). You can feel free to not link to such basic infra terms if you think it's silly though; that's up to editor discretion.